Terminal.Gui/Views/ColorBar.cs

```diff
@@ -47,6 +47,13 @@ public int Value
         }
     }
 
+    /// <inheritdoc />
+    public void SetValueWithoutRaisingEvent (int v)
```

Comment:
Not a fan of this.

If setting something without raising events is necessary, unsubscribe.

What if the user _wants_ those events to be raised for every individual value for whatever reason?

Otherwise, this should be private.

Reply:
I added this only because @tig didn't like `bool updating` approach in parent view - see https://github.com/gui-cs/Terminal.Gui/pull/3604#discussion_r1677164640

We need a way to sync all the sub bits of ColorPicker without getting circular events being triggered.

A compromise is perhaps that the method can be made internal?

Reply:
Another suggestion is add a bool property like RaiseEventForAllValues.

Reply:
The point of the change was to remove variables and before/after operations.

We do want the event raised for clicks and keyboard changes just not when we are sync a change e.g. when user tupes new value in hex box

Reply:
Sorry, I interacted on impulse due to @dodexahedron saying that the user might want to be notified of all changes. Really, if they are unnecessary, avoiding them from generating events is the best option.

Reply:
I have hidden the method and the class as both are internal sub components of ColorPicker.

See  [21897b5](https://github.com/gui-cs/Terminal.Gui/pull/3604/commits/21897b5cba3bb3df168f63a1c9511abe5dc08463)

Is this acceptable compromise @dodexahedron?

Reply:
> I added this only because @tig didn't like `bool updating` approach in parent view - see [#3604 (comment)](https://github.com/gui-cs/Terminal.Gui/pull/3604#discussion_r1677164640)
> 
> We need a way to sync all the sub bits of ColorPicker without getting circular events being triggered.
> 
> A compromise is perhaps that the method can be made internal?

Yeah even internal events are a possibility. I haven't given it additional dedicated thought, yet, so at the moment it's just brainstorming essentially.

Reply:
> The point of the change was to remove variables and before/after operations.
> 
> We do want the event raised for clicks and keyboard changes just not when we are sync a change e.g. when user tupes new value in hex box

Well... Why not?

Other color pickers I (unreliably) recall usually update in real time.

Paint.net definitely does. Observe:

https://github.com/user-attachments/assets/565cf7bc-606a-4f6a-ac27-cb3f458167c7

However, making that behavior work exactly like that, for all possible 1-8 digit hex strings, would probably require some expansion of the color parse switch block to handle the 1, 2, 4, 5, and 7 character cases, I think. ...Though it's been many moons since I wrote that, so I don't remember all of the cases it handles off the top of my head. I'm pretty sure those are the cases not covered.

Alternatively/more simply, we could have the parse code only raise the event for strings that are meaningful, like 1 (interpret as that nibble repeated 6 times), 2 (that byte repeated 3 times), 3 (it already does that - each nibble is doubled, like on the web), 4 (also does this I'm pretty sure - same as 3 but with alpha), 6 (of course already does this), and 8 (again, already does this). So that only really means adding cases for 1 and 2, I guess? I suppose 5 and 7 _could_ be handled, assuming there's just an omitted leading zero for both, making them just use the 6 and 8 character cases... 🤷‍♂️ 

And of course that struct is natively ARGB, but that's only relevant for naming really because it's just a 32-bit binary value. So, following some of my original thoughts farther down the rabbit hole, I think the way I might end up dealing with that is pretty much how .net 8 and the System.Numerics pseudo-recursive interfaces with static virtuals work: A couple of common interfaces with things like the named colors, and then ColorBlahBlah structs for each pixel format/color space, so they're largely interchangeable, the way things like `INumber<TSelf>` are, regardless of the underlying struct[^IfIKnewThenWhatIKnowNow].

That's where the things like int.Zero come from - they're defined on the static virtual interface member, and then each implementing type brings its own implementation of that member, but it is static, so it is available on the _type argument_ of the interface (so, the color types themselves), meaning it's a largely non-breaking change, insofar as Color itself and its visible API surface goes. 

Anyway... There's what occurred to me while thinking about it right now. Plenty of options available other than those, of course, but I think those would be very low-code and highly idiomatic C#12+ ways to go about it, at least, with details up to the implementation and ultimately irrelevant to the consumer (yay for interfaces!).

[^IfIKnewThenWhatIKnowNow]: Static virtuals on self-referencing generic interfaces are honestly something we should do to Dim/Pos to polish them up and make them seamless, as originally intended when they were nested types hidden from the consumer. They're exactly the solution to it all. When that stuff was happening, I hadn't yet had a chance to get terribly intimate with that language feature yet. I've come to love it because it really does deal with that exact scenario so well. Sure do wish I were familiar with it months ago, when we were trying to figure that all out. It is a bit tricky to initially get the hang of writing because it's VERY particular about the syntax, but it's all strongly-typed static analysis, so you pretty much can't write it wrong and manage to compile it. Score another point for Roslyn.

Reply:
You know... Come to think of it...

Paint.Net is licensed with an MIT equivalent, just with an explicit trademark exception (so they should have just used MPL-2.0, but oh well).

So, we could even go so far as to lift the code for that entire control and translate to the Terminal.Gui API, and all we need is a license notice, just like any other dependency already requires (which we need to be sure we shore up in the nupkg before release, btw).

Reply:
It already behaves as paint.net one. I.e. real-time sync.

The code here is just to avoid infinite sync looping

Reply:
A fix should be pretty quick I think but I'm on my way out the door so I'll look later. It's not a huge deal as a way to, at the end of the day, achieve the same result. But I would request that method not be public, please, so nobody uses it before it gets polished up. Or drop an obsolete attribute on it. Or whatev. 🤷‍♂️ Or just @mention me in a todo in the code if you don't wanna mess with it and I'll include it in one of my ADHD ~~nitpick~~ cleanup passes. 😅

Reply:
> But I would request that method not be public

I already made it not public in https://github.com/gui-cs/Terminal.Gui/pull/3604/commits/21897b5cba3bb3df168f63a1c9511abe5dc08463